server/channels/app/team.go

```diff
@@ -510,14 +510,18 @@ func (a *App) UpdateTeamMemberSchemeRoles(c request.CTX, teamID string, userID s
 		return nil, err
 	}
 
-	member.SchemeAdmin = isSchemeAdmin
-	member.SchemeUser = isSchemeUser
-	member.SchemeGuest = isSchemeGuest
+	if member.SchemeGuest {
+		return nil, model.NewAppError("UpdateTeamMemberSchemeRoles", "api.team.update_team_member_roles.guest.app_error", nil, "", http.StatusBadRequest)
+	}
 
-	if member.SchemeUser && member.SchemeGuest {
+	if isSchemeUser && isSchemeGuest {
```

Comment:
What happens if I pass `isSchemeAdmin` and `isSchemeGuest` as true but `isSchemeUser` as false?

Reply:
The above if statement should be met and the request rejected with a bad request

Reply:
Oh wait not really as is not looking for the flags but the fact that the user was already a guest.

I guess that would be allowed and it probably shouldn't.

I can add that too... thanks @jespino